pkg/maya/maya.go

```diff
@@ -27,22 +29,74 @@ import (
 	mach_apis_meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// MayaPlaceholders is a structure that is composed
+// of various properties. These properties are set as
+// placeholders in a template file.
+type MayaPlaceholders struct {
+	// Owner represents the name of the owner
+	Owner string `json:"owner"`
+}
+
 // MayaYaml represents the yaml definition
 // that is typically embedded in various other
 // Maya types
+//
+// MayaYaml is expected to be marshalled into corresponding
+// go structure.
 type MayaYaml struct {
-	// Yaml represents the yaml in string format
+	// Yaml represents a yaml in string format. This string
+	// formatted yaml acts as a template.
 	Yaml string
+
+	// MayaPlaceholders represents various placeholders
+	// that might have been set in above Yaml property.
+	MayaPlaceholders
+
+	// YmlBytes represents above yaml in byte slice format
+	//
+	// NOTE:
+	//  This is generated by using the templated yaml
+	// (i.e. Yaml property) & placeholders (i.e.
+	// MayaPlaceholders property)
+	YmlBytes []byte
```

Comment:
@AmitKumarDas , I couldn't understand how this gets enabled on UT. The variable m* MayaYaml is a global variable under package maya which has redundant piece of variables - Yaml string and YmlBytes []byte. Can u pls explain?

Reply:
Thanks @prateekpandey14 @gkGaneshR for the reviews. There are 3 properties in MayaYaml struct. Bytes is derived from placeholders & string version of yaml. In some of the cases bytes are derived from string yaml only. However, all these explanation still begs the question on why bytes, as it is redundant.

To answer it, let us think from object point of view. There are cases where the object i.e. MayaYaml needs to be initialized out of bytes directly.  In these cases what should be the struct design?

Let us not think only from a behavior p.o.v. What about object construction & object composition?
Thinking only from behavior p.o.v will lead to procedural code that fails the open-closed principle.

MayaYaml is constructed in 2 ways.
Way 1 - by providing placeholder & string yaml e.g. Load() of MayaConfigMap
Way 2 - by providing bytes[] only e.g. NewMayaServiceByByte()

If you look at the _test.go code, you will find the motive for doing this.